docs/source/reference/envvars.rst

```diff
@@ -35,6 +35,23 @@ for permanent use by adding::
 
     color_scheme: dark_bg
 
+Jit flags
+---------
+
+These variables globally override flags to the :func:`~numba.jit` decorator.
```

Comment:
It would be extremely surprising to me if I explicitly wrote `boundscheck=False` and that didn't have any effect because of a hidden environment variable `NUMBA_BOUNDSCHECK=1` which I didn't even know about.

I guess you want to give a user the ability to modify the default behaviour of a library they didn't write so maybe it's unavoidable?

Reply:
Yes, that's the idea. The idea is to add a test build for numba itself with it globally enabled, but I can definitely see this being useful for general code that uses numba. If you get weird results, you can turn this on to see if its an out of bounds access. You can get the same thing by disabling jit entirely (`NUMBA_DISABLE_JIT`), but sometimes that's too slow to be tenable. I didn't realize this would be controversial. 

I can make it so that explicitly writing `jit(boundscheck=False)` overrides even the environment variable. That would be a rather rare thing to write, since the default is already False. But that also makes the overall logic for the flag and the environment variable a bit confusing IMO.

Reply:
I think it's fine as-is - just something  for a user to watch out for. Assuming they're not sharing a computer they would have had to set the env var themselves so they should know about it...